src/strategy/mean_calculation.rs

```diff
@@ -0,0 +1,61 @@
+use std::collections::VecDeque;
+
+use serde::{ Deserialize, Serialize };
+
+#[derive(Debug, Serialize, Deserialize, Clone, Copy)]
+pub enum MeanCalculationMethod {
+    SimpleMA,
+    EMA,
+    VMAP,
+}
+
+pub struct MaTracker {
+    method: MeanCalculationMethod,
+    period: usize,
+    values: VecDeque<f64>,
+    sum: f64,
+}
+
+pub trait MeanCalculation {
+    fn update(&mut self, value: f64) -> f64;
+
+    fn calculate(&self) -> f64;
+}
+
+impl MaTracker {
+    pub fn new(period: usize, ma_calculation_method: MeanCalculationMethod) -> Self {
+        Self {
+            method: ma_calculation_method,
+            period,
+            sum: 0_f64,
+            values: VecDeque::new(),
+        }
+    }
+}
+
+impl MeanCalculation for MaTracker {
+    fn update(&mut self, value: f64) -> f64 {
+        self.sum += value;
```

Comment:
This is good enough for mid frequency, but there is better method which is used across the board. Instead of keeping whole history you can use exponential moving average, which only requires previous calculated value and new value. VecDeque would require allocations which is not good in HFT,  though it is not a first priority optimization. Ema is calculated as `acc * alpha + new_value * (1 - alpha)`, where acc is accumulator. Alpha is decay factor and is 2 /(1+span)